component/common/appendable/appendable.go

```diff
@@ -33,10 +37,15 @@ type flowAppender struct {
 func (app FlowAppendable) Appender(_ context.Context) storage.Appender {
 	return &flowAppender{
 		buffer:    make(map[int64][]*metrics.FlowMetric),
-		receivers: app,
+		receivers: app.receivers,
```

Comment:
I'm pretty sure you need to protect this with a mutex now to prevent race conditions between updating the component and scraping happening at the same time

Reply:
Probably a RWMutex since many scrape loops might want a rear lock at roughly the same time